### PR TITLE
[Test] Move archived setting test into its own test class

### LINKED ISSUE #111774: [CI] CoreFullClusterRestartIT testSnapshotRestore {cluster=OLD} failing
**Build Scans:**
- [elasticsearch-intake #12309 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mx3sewq36c6yi)
- [elasticsearch-pull-request #39234 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ryuuzds5cuia6)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:full-cluster-restart:v8.15.4#bwcTest" -Dtests.class="org.elasticsearch.xpack.restart.CoreFullClusterRestartIT" -Dtests.method="testSnapshotRestore {cluster=OLD}" -Dtests.seed=4EF8F35CEACABC20 -Dtests.bwc=true -Dtests.locale=ff-Latn-LR -Dtests.timezone=America/Ensenada -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.restart.CoreFullClusterRestartIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSnapshotRestore%20%7Bcluster%3DOLD%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (<[8.15.4]> or <[8512000]> or <[8.15.0-8.15.3]>)
     but: was <[8.15.0-8.15.4]>
```

**Issue Reasons:**
- [main] 2 consecutive failures in step 8.15.4_bwc-snapshots
- [main] 2 failures in test testSnapshotRestore {cluster=OLD} (33.3% fail rate in 6 executions)
- [main] 2 failures in step 8.15.4_bwc-snapshots (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### LINKED ISSUE #111777: [CI] FullClusterRestartIT testSnapshotRestore {cluster=OLD} failing
**Build Scans:**
- [elasticsearch-intake #12309 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mx3sewq36c6yi)
- [elasticsearch-pull-request #39435 / 8.15.4_bwc](https://gradle-enterprise.elastic.co/s/3elzax34b45pg)
- [elasticsearch-pull-request #39435 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/b3nqk7grkdbsy)
- [elasticsearch-pull-request #39234 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ryuuzds5cuia6)

**Reproduction Line:**
```
./gradlew ":qa:full-cluster-restart:v8.15.4#bwcTest" -Dtests.class="org.elasticsearch.upgrades.FullClusterRestartIT" -Dtests.method="testSnapshotRestore {cluster=OLD}" -Dtests.seed=4EF8F35CEACABC20 -Dtests.bwc=true -Dtests.locale=yue-CN -Dtests.timezone=America/Danmarkshavn -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.upgrades.FullClusterRestartIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSnapshotRestore%20%7Bcluster%3DOLD%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (<[8.15.4]> or <[8512000]> or <[8.15.0-8.15.3]>)
     but: was <[8.15.0-8.15.4]>
```

**Issue Reasons:**
- [main] 2 consecutive failures in step 8.15.4_bwc-snapshots
- [main] 4 failures in test testSnapshotRestore {cluster=OLD} (16.0% fail rate in 25 executions)
- [main] 2 failures in step 8.15.4_bwc-snapshots (100.0% fail rate in 2 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### LINKED ISSUE #111798: [CI] FullClusterRestartIT testSnapshotRestore {cluster=UPGRADED} failing
**Build Scans:**
- [elasticsearch-intake #8871 / 7.17.24_bwc-snapshots](https://gradle-enterprise.elastic.co/s/au6jsxfwxb2ia)
- [elasticsearch-intake #8870 / 7.17.24_bwc-snapshots](https://gradle-enterprise.elastic.co/s/pz33nf4nqmaei)

**Reproduction Line:**
```
./gradlew ":qa:full-cluster-restart:v7.17.24#bwcTest" -Dtests.class="org.elasticsearch.upgrades.FullClusterRestartIT" -Dtests.method="testSnapshotRestore {cluster=UPGRADED}" -Dtests.seed=12602FAD7BA29740 -Dtests.bwc=true -Dtests.locale=ro -Dtests.timezone=Asia/Riyadh -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testSnapshotRestore%20%7Bcluster%3DUPGRADED%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.upgrades.FullClusterRestartIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:32809], URI [/info/_doc/testsnapshotrestore_count?filter_path=_source], status line [HTTP/1.1 404 Not Found]
{}
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testSnapshotRestore {cluster=UPGRADED}
- [main] 2 consecutive failures in step 7.17.24_bwc-snapshots
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testSnapshotRestore {cluster=UPGRADED} (0.6% fail rate in 358 executions)
- [main] 2 failures in step 7.17.24_bwc-snapshots (8.7% fail rate in 23 executions)
- [main] 2 failures in pipeline elasticsearch-intake (20.0% fail rate in 10 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### LINKED ISSUE #111799: [CI] CoreFullClusterRestartIT testSnapshotRestore {cluster=UPGRADED} failing
**Build Scans:**
- [elasticsearch-intake #12313 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4gktpimoqifye)
- [elasticsearch-intake #12309 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/wvgt4wwhitkii)
- [elasticsearch-pull-request #39580 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/hqxvqx4kxnoxa)
- [elasticsearch-pull-request #39580 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/x3kdybd6dw4he)
- [elasticsearch-intake #12313 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/u4x7fodlx6mwy)
- [elasticsearch-intake #12313 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/istklvemrlzvg)
- [elasticsearch-intake #12309 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mx3sewq36c6yi)
- [elasticsearch-intake #12309 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/7pbnt52e27c3q)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:full-cluster-restart:v8.17.0#bwcTest" -Dtests.class="org.elasticsearch.xpack.restart.CoreFullClusterRestartIT" -Dtests.method="testSnapshotRestore {cluster=UPGRADED}" -Dtests.seed=62CA000E87452EBD -Dtests.bwc=true -Dtests.locale=hi-Latn-IN -Dtests.timezone=Antarctica/Rothera -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.restart.CoreFullClusterRestartIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSnapshotRestore%20%7Bcluster%3DUPGRADED%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
org.elasticsearch.client.ResponseException: method [PUT], host [http://[::1]:39773], URI [/_cluster/settings], status line [HTTP/1.1 400 Bad Request]
---
error:
  root_cause:
  - type: "illegal_argument_exception"
    reason: "unknown setting [archived.cluster.routing.allocation.balance.threshold]\
      \ did you mean any of [cluster.routing.allocation.balance.threshold, cluster.routing.allocation.balance.shard]?"
  type: "illegal_argument_exception"
  reason: "unknown setting [archived.cluster.routing.allocation.balance.threshold]\
    \ did you mean any of [cluster.routing.allocation.balance.threshold, cluster.routing.allocation.balance.shard]?"
status: 400

```

**Issue Reasons:**
- [main] 2 consecutive failures in test testSnapshotRestore {cluster=UPGRADED}
- [main] 2 consecutive failures in step 8.17.0_bwc-snapshots
- [main] 3 consecutive failures in step 8.15.4_bwc-snapshots
- [main] 3 consecutive failures in step 8.16.0_bwc-snapshots
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 8 failures in test testSnapshotRestore {cluster=UPGRADED} (80.0% fail rate in 10 executions)
- [main] 2 failures in step 8.17.0_bwc-snapshots (100.0% fail rate in 2 executions)
- [main] 3 failures in step 8.15.4_bwc-snapshots (100.0% fail rate in 3 executions)
- [main] 3 failures in step 8.16.0_bwc-snapshots (100.0% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-intake (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### PR DESCRIPTION
**Pull request by ywangd**
As title says, the archived setting needs to be moved out to not interfere with other test methods in the same test class.

Resolves: #111798
Resolves: #111777
Resolves: #111774
Resolves: #111799

### CODE REVIEW

**Comment by ywangd** [comment]
This PR should definitely fixes the failures with the upgraded cluster, i.e. #111799 and #111798
The failures with the old cluster (#111774, #111777) are caused by version bump fallouts. I believe these are no longer issues. So this PR unmutes all of them.

**Comment by ywangd** [comment]
Oh joy. Looks like the fix did not work. Let me take another look.

**Comment by ywangd** [comment]
I pushed more changes for the fix 6eaca3c21f4b3d06b94ce0fa285f8bd0a37131ed
* It is not sufficient to just remove the archive settings at the end of test because other tests may run first in the upgraded cluster and fail. So I made the test for archive setting to always run first.
* The version number fallout for 8.15 is still an issue. I think the automated commit (b8688b38c8728540fc91ff6833b793c222ac0654) missed 8.15.4 in the csv files. I manually added them. Please let me know if this needs to be done differently.

File: test/framework/src/main/java/org/elasticsearch/test/AnnotationTestOrdering.java [hunk 1]
```diff
@@ -25,10 +26,11 @@
...
-        return Integer.compare(
-            o1.getTestMethod().getAnnotation(Order.class).value(),
-            o2.getTestMethod().getAnnotation(Order.class).value()
-        );
+        return Integer.compare(getOrderValue(o1.getTestMethod()), getOrderValue(o2.getTestMethod()));
+    }
+
+    private int getOrderValue(Method method) {
+        return method.isAnnotationPresent(Order.class) ? method.getAnnotation(Order.class).value() : Integer.MAX_VALUE;
     }
```
Comment (ywangd) [inline test/framework/src/main/java/org/elasticsearch/test/AnnotationTestOrdering.java:34]: This maybe controversial. It essentially allows annotating `Order` for a subset of tests instead of all the tests. Missing Order annotation defaults to Integer.MAX_VALUE, i.e. they run last. I am trying to avoid annotating every method in `FullClusterRestartIT`. It is really needed for just one method. The rest can run in whatever order. I also considered splitting the single method into a different test class. But `CoreFullClusterRestartIT` extends `FullClusterRestartIT` which means we need yet another test class to extends the new test class. Seems not worth it. So I went with this approach.

**Review by idegtiarenko — APPROVED** [review]

File: test/framework/src/main/java/org/elasticsearch/test/AnnotationTestOrdering.java (hunk 1 above)
Comment (ywangd) [reply test/framework/src/main/java/org/elasticsearch/test/AnnotationTestOrdering.java]: Nevermind. I just noticed the full-cluster-restart package has its own `FullClusterRestartTestOrdering` annotation. So I have moved the changes into it which should be less of an issue. The tests also need to respect the ordering determined by old and new cluster as well.

**Comment by ywangd** [comment]
@DaveCTurner Now I am getting a new [failure](https://gradle-enterprise.elastic.co/s/ok6q6lqkcqrro/tests/task/:x-pack:qa:full-cluster-restart:v8.15.4%23bwcTest/details/org.elasticsearch.xpack.restart.CoreFullClusterRestartIT/testSnapshotRestore%20%7Bcluster%3DUPGRADED%7D?expanded-stacktrace=WyIwIl0&top-execution=1) which I suspect is a legitimate issue. 

The important bit of the exception is
```
[2024-11-08T04:54:27,351][WARN ][o.e.s.RestoreService     ] [test-cluster-0] [repo:old_snap/ZwUuFTIlQ1-qeTsp8Cehcg] failed to restore snapshot java.lang.IllegalArgumentException: illegal value can't update [cluster.routing.allocation.balance.threshold] from [1.0] to [0.999]
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.common.settings.Setting$Updater.getValue(Setting.java:1304)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.common.settings.AbstractScopedSettings.validateUpdate(AbstractScopedSettings.java:139)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.snapshots.RestoreService$RestoreSnapshotStateTask.applyGlobalStateRestore(RestoreService.java:1546)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.snapshots.RestoreService$RestoreSnapshotStateTask.execute(RestoreService.java:1477)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:573)
```

IIUC, this means users cannot restore a snapshot taken on an old cluster if it has a less than 1.0 value for persistent setting `cluster.routing.allocation.balance.threshold`. While we allow index settings to be ignored during restore, the same thing does not seem to be available for cluster settings. The workaround is to _not_ restore cluster state which may not always be acceptable. Or tempering with on disk metadata snapshot which is not optimal either. We may want to reconsider whether the strict validation is viable in 9.0?

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
Ah this is unfortunate. I think my preferred solution would be to run this specific test case in a separate Gradle project so it does not interfere with these other tests.

In the meantime, I'd suggest muting the problematic test case and unmuting the ones that were muted in error.

**Comment by ywangd** [comment]
Once `testBalancedShardsAllocatorThreshold` is separated out in its own test class. All tests should run successfully. So I'll leave all of them running and raise a separate issue for the strict validation.

**Comment by ywangd** [comment]
Raised https://github.com/elastic/elasticsearch/issues/116558 for the validation issue

File: qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/FullClusterRestartArchivedSettingsIT.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import static org.elasticsearch.cluster.routing.allocation.allocator.BalancedShardsAllocator.THRESHOLD_SETTING;
+
+/**
+ * Tests to run before and after a full cluster restart. This is run twice,
+ * one with {@code tests.is_old_cluster} set to {@code true} against a cluster
+ * of an older version. The cluster is shutdown and a cluster of the new
+ * version is started with the same data directories and then this is rerun
+ * with {@code tests.is_old_cluster} set to {@code false}.
+ */
+public class FullClusterRestartArchivedSettingsIT extends ParameterizedFullClusterRestartTestCase {
```
Comment (ywangd) [inline qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/FullClusterRestartArchivedSettingsIT.java:39]: I didn't add a subclass for this new test class in x-pack because it does not seem necessary. The archived settings do not depend on whether security is enabled. But please let me know if you think have one to pair the coverage.

File: qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/FullClusterRestartArchivedSettingsIT.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        final var settingsPath = "persistent." + THRESHOLD_SETTING.getKey();
+        final var settingValue = clusterSettingsResponse.evaluate(settingsPath);
+
+        if (isRunningAgainstOldCluster()) {
+            assertEquals(chosenValue, settingValue);
+        } else {
+            assertNull(settingValue);
+            assertNotNull(clusterSettingsResponse.<String>evaluate("persistent.archived." + THRESHOLD_SETTING.getKey()));
+            // Remove the archived setting to not interfere other tests in this test class
+            client().performRequest(
```
[the remark this answers is not in the document]
Comment (ywangd) [reply qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/FullClusterRestartArchivedSettingsIT.java]: This is a left-over from the previous change. Thanks for spotting it. I pushed 3b504752052c1e54e209ea59acf0d187a80760b0 to remove them.

File: server/src/main/resources/org/elasticsearch/index/IndexVersions.csv [hunk 4]
```diff
@@ -131,3 +131,4 @@
 8.15.1,8512000
 8.15.2,8512000
 8.15.3,8512000
+8.15.4,8512000
```
[the remark this answers is not in the document]
Comment (ywangd) [reply server/src/main/resources/org/elasticsearch/index/IndexVersions.csv:134]: Yes. Otherwise the test would fail with a message similar to the following
```
java.lang.AssertionError: 
Expected: (<[8.14.4]> or <[8512000]> or <[8.14.0-8.14.4]>)
     but: was <[8.14.0-8.14.3]>
```
See #111777 for an example. 

@mark-vieira Could you please take a look at this change and make sure it is OK to update it manually like this?

File: server/src/main/resources/org/elasticsearch/TransportVersions.csv [hunk 5]
```diff
@@ -131,3 +131,4 @@
 8.15.1,8702002
 8.15.2,8702003
 8.15.3,8702003
+8.15.4,8702003
```
[the remark this answers is not in the document]
Comment (ywangd) [reply server/src/main/resources/org/elasticsearch/TransportVersions.csv:134]: Probably not strictly necessary for the tests for pass. But it would be rather odd if we update `IndexVersions.csv` without the corresponding change here.

File: server/src/main/resources/org/elasticsearch/index/IndexVersions.csv (hunk 4 above)
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/resources/org/elasticsearch/index/IndexVersions.csv:134]: I think I understand why this change needs to be made, I just don't understand why it's happening in this PR. Shouldn't this be added by the release process?

File: muted-tests.yml [hunk 6]
```diff
@@ -156,15 +153,9 @@ tests:
...
-  issue: https://github.com/elastic/elasticsearch/issues/111799
-- class: org.elasticsearch.xpack.restart.CoreFullClusterRestartIT
-  method: testSnapshotRestore {cluster=OLD}
-  issue: https://github.com/elastic/elasticsearch/issues/111774
-- class: org.elasticsearch.upgrades.FullClusterRestartIT
-  method: testSnapshotRestore {cluster=OLD}
-  issue: https://github.com/elastic/elasticsearch/issues/111777
+- class: org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT
+  method: testGeoPointGeoTile
+  issue: https://github.com/elastic/elasticsearch/issues/115818
```
Comment (DaveCTurner) [inline muted-tests.yml:158]: This also seems unrelated - should this test be muted in `main` instead?

Comment (ywangd) [reply muted-tests.yml]: Must be a fallout from resolving merge conflict. Removed in 0713e01456b05a3e5df96f99c47a0d289408a5cf

File: server/src/main/resources/org/elasticsearch/index/IndexVersions.csv (hunk 4 above)
[the remark this answers is not in the document]
Comment (ywangd) [reply server/src/main/resources/org/elasticsearch/index/IndexVersions.csv:134]: > Shouldn't this be added by the release process?

I have the same question. That's reason I requested @mark-vieira 's review. Maybe I should have tagged @elastic/es-delivery 
Also ping @thecoop since #111777 is assigned to you. Thank you!

[the remark this answers is not in the document]
Comment (thecoop) [reply server/src/main/resources/org/elasticsearch/index/IndexVersions.csv:134]: The build automation I think is not quite working properly with the multiple release branches we have - there's also https://github.com/elastic/elasticsearch/issues/114972. I need to look at this properly, but these should be added in a separate PR so we can confirm that it is actually the right thing to do at the right time (and backport it appropriately too)

[the remark this answers is not in the document]
Comment (ywangd) [reply server/src/main/resources/org/elasticsearch/index/IndexVersions.csv]: Thanks @thecoop !
I also noticed the missing versions are now added in main 85b2bab2e38409449a37adc9408902cbf79f8c8f
So these changes are indeed obsolete. I have removed them with e14dcb022c8ca159518beb7f80585b7f5bf5cc5a

[the remark this answers is not in the document]
Comment (ywangd) [reply server/src/main/resources/org/elasticsearch/index/IndexVersions.csv]: Errr the tests now [complain](https://gradle-enterprise.elastic.co/s/7rwqwan4xgxki/tests/task/:qa:full-cluster-restart:v8.16.1%23bwcTest/details/org.elasticsearch.upgrades.FullClusterRestartIT/testSnapshotRestore%20%7Bcluster=OLD%7D?top-execution=1) about 8.16.1 which is not yet in the versions file. 😿 

```
Expected: (<[8.16.1]> or <[8518000]> or <[8.16.0]>) |  
    but: was <[8.16.0-8.16.1]>
```

Cannot really move this PR forward without the versions to be fixed first.

File: muted-tests.yml (hunk 6 above)
Comment (mark-vieira) [reply muted-tests.yml]: @DaveCTurner you are correct. It's updated by our release automation.

https://github.com/elastic/elasticsearch/commit/56bc6fda6f342a435837f0eef378fd96f70cab5a#diff-84af7e2f9a12f89edad6ecb30bbb1f60b45a285b2fdcec8b62d7dfc6ae64d6ee

**Review by mark-vieira — APPROVED** [review]

**Review by DaveCTurner — APPROVED** [review]
LGTM2

I expect the bwc test failures relate to the recent release of 8.16.0 - they should be addressed in `main` (soon, if not already, I haven't checked) and then this can be merged in its current state.

**Comment by thecoop** [comment]
The version check failure has been resolved by https://github.com/elastic/elasticsearch/pull/116727

**Comment by ywangd** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/116902)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)
